Add 2019 Honda Insight

theantihero opened the pull request:
Related to closed PR #417 due to fp issue. Was able to get fw ids from the query script.

pd0wm commented:
Do you have a route we can use for the regression tests? Please put one in test_car_models with a fixed fingerprint: like https://github.com/commaai/openpilot/blob/master/selfdrive/test/test_car_models.py#L293

pd0wm commented:
And you also need to add the fingerprint to the list of fingerprints since that is still used in the `is_ecu_disconnected` fingerprint. To make sure it doesn't interfere with other cars fingerprinting add a list with `IGNORED_FINGERPRINTS` like https://github.com/commaai/openpilot/blob/master/selfdrive/car/toyota/values.py#L269

theantihero commented:
@pd0wm Hey, thanks for the additional comments, I added the actual fp and added it to ignored list.
I will provide a route later today. Does it have to be a stock repo drive or can a dirty fork be used?

pd0wm commented:
The route doesn’t matter

pd0wm commented:
Pushed a fix to the fingerprint test to master:https://github.com/commaai/openpilot/commit/6eca88e9bcc4c6d32304444c6565ca6be0d77bbb

Can you rebase?

pd0wm commented on selfdrive/test/test_car_models.py:
```diff
@@ -191,6 +191,10 @@ def get_route_log(route_name):
+  },
+    "fb51d190ddfd8a90|2020-02-25--14-43-43": {
+    'carFingerprint': HONDA.INSIGHT,
+    'enableCamera': True,
```
Add `'fingerprintSource': 'fixed'`

theantihero replied:
Will do

pd0wm commented on selfdrive/car/honda/values.py:
```diff
@@ -651,6 +673,7 @@ class CAR:
   CAR.PILOT: dbc_dict('honda_pilot_touring_2017_can_generated', 'acura_ilx_2016_nidec'),
   CAR.PILOT_2019: dbc_dict('honda_pilot_touring_2017_can_generated', 'acura_ilx_2016_nidec'),
   CAR.RIDGELINE: dbc_dict('honda_ridgeline_black_edition_2017_can_generated', 'acura_ilx_2016_nidec'),
+  CAR.INSIGHT: dbc_dict('honda_insight_ex_2019_can_generated.dbc', None),
```
Don't add .dbc exension. This is why the test hangs.

theantihero replied:
Ah.. Happy little accident, my mistake.
